frontend/src/properties/Group.tsx

```diff
@@ -76,6 +77,8 @@ const widgetToPropertySection = (
     .filter((subscription) => subscription.widgetId === widget.id)
     .forEach((subscription) => subscribers.push(...subscription.subscribers));
 
+  const hasDiagnostic = containsDiagnostic(widget);
```

Comment:
You don't need to compute something that simple beforehand and thus modify the API of all the property sections needlessly. This can and should be done in each property section. We are just talking about checking if an array is empty.